### ISSUE TITLE
Add Minder

### ISSUE THREAD

**Issue by aeghn** [msys2/MINGW-packages#16392]
### Package name

Minder-git

### Brief description of package

Mind-mapping application for Elementary OS.

![图片](https://user-images.githubusercontent.com/51843252/227428077-3fe03b57-2e38-4bd0-b386-ce6bb87b712b.png)

### URL for package's homepage

https://github.com/phase1geo/Minder

### Provide a basic test case to validate the package's functionality.

Open it and add some nodes.

![图片](https://user-images.githubusercontent.com/51843252/227429023-bfb0bec0-ca57-4210-9787-025ca118de34.png)

### MINGW environments where you need the package

- [X] MINGW64
- [X] MINGW32
- [X] UCRT64
- [X] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Are you willing to submit a PR?

yes, and actually I have running it on my pc https://github.com/aeghn/MINGW-packages/commit/b3158ba5bbd13fd8fdedb379e78e3f3ac86bff35

**Comment by aeghn** [comment]
But I have some problems now, I cannot make the input method bar focus in the node. I think it is because the node is implemented by this software.

I'm trying to solve this problem, then I will submit a merge request.

![图片](https://user-images.gitubusercontent.com/51843252/227429383-45b0008d-5381-45f7-8489-5c920e4514dc.png)

It seems like 
```vala
    _im_context.set_cursor_location( rect );
```
https://github.com/phase1geo/Minder/blob/fbf808e5b2b3d4e907f77405c5d236cba279252c/src/DrawArea.vala#L920

didn't work.

And ideas? Thanks in advance.

**Comment by Biswa96** [comment]
Some points:
* Any issue should be reported to upstream first.
* Patches should be submitted to upstream first.
* The package should be built from tag release, not from git repository.

**Comment by aeghn** [comment]
>     * Any issue should be reported to upstream first.

There are some Windows's specific problems, so I think maybe the gays in this community have more experiences.

>     * Patches should be submitted to upstream first.

They are some patches I think which should only to be patched for the windows.

>     * The package should be built from tag release, not from git repository.

There are some issues in saving files on Windows with the latest release, so I use the git version.


Anyway, thanks for you reply. @Biswa96

**Comment by Biswa96** [comment]
Does upstream even support Windows platform?

**Comment by aeghn** [comment]
> I don't know a lot about GTK, but the "IME candidate window not following the cursor" issue is because the UI framework does not support Windows IME framework (IMM or TSF) properly. A search of "GTK Windows TSF" and "GTK Windows IME" came up empty, so I think it will be a major issue to resolve.

Yes, the main issue is DrawArea implemented its own im_context, I think it is the main issue, I'm tring to understand the TextArea source code, but I didn't find the key to this problem. 

> Do we have other GTK app in MSYS2 to test this?

Akira? 

![图片](https://user-images.githubusercontent.com/51843252/227925995-a70c5895-b767-4a76-a725-0f53842785c5.png)

But I find I can not  edit `Text` in Akira, even in English.

![图片](https://user-images.githubusercontent.com/51843252/227929069-af40f623-42b2-4249-8e53-971bef19ae84.png)


![图片](https://user-images.githubusercontent.com/51843252/227927501-b4877fb0-5a98-4f98-b097-f788108cb681.png)

**Comment by driver1998** [comment]
Looks like you'll need to figure out how to handle IME properly in GTK first. Maybe ask around in their [forum](https://discourse.gnome.org/c/platform/core/9).

**Comment by aeghn** [comment]
After some test, I found it is 

```
    _im_context.set_client_window( this.get_window() );
```

which doesn't work, because `this.get_window()` didn't work on Windows, I will try to fix it.
